app/models/payment/go_cardless/transaction.rb

```diff
@@ -81,8 +81,10 @@ def publish_failed_subscription_charge
     ChampaignQueue.push({
       type: 'subscription-payment',
       params: {
+        created_at: created_at,
         recurring_id: subscription.go_cardless_id,
-        success: 0
+        success: 0,
+        status: 'failed'
```

Comment:
I agree! I feel like we use the worker like a proxy to AK, managing the AK api specifics from Champaign. Ideally Champaign should only care about publishing events, and the worker should interface with AK and handle any specific logic and syntax needed by their API.